test/bin/manage_build_cache.sh

```diff
@@ -0,0 +1,169 @@
+#!/bin/bash
+#
+# This script should be run on the build host to manage cache of build artifacts
+set -euo pipefail
+
+SCRIPTDIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+# shellcheck source=test/bin/common.sh
+source "${SCRIPTDIR}/common.sh"
+
+AWS_BUCKET_NAME="${AWS_BUCKET_NAME:-microshift-build-cache}"
+RELEASE_SUBDIR=
+BUILDID_SUBDIR=
+
+usage() {
+    cat - <<EOF
+${BASH_SOURCE[0]} (upload|download|exist|latest|keep) [options]
```

Comment:
Can you show examples of actual invocation of the script?

Reply:
```
manage_build_cache.sh upload -r 4.15 -b 231030
manage_build_cache.sh download -r 4.15 -b latest

if manage_build_cache.sh exist -r 4.15 -b latest &>/dev/null; then
   build_opts=-I # disable ISO builds
fi 

manage_build_cache.sh latest -r 4.15 -b 231031
```

Reply:
how will presubmit know which tag to use?
could it pick partial image? e.g. when one job starts pushing tagX/a.iso, could another job see tagX as existing?

Reply:
OK, there are good questions:
- I think we pre-submit always need to use `latest` tag
- Since the bucket operations are atomic, it cannot pick a partial image. A file becomes available only after its upload is finished.
- The question is how to implement an atomic rename/copy to latest.

Since we would have a check for the cache validity, maybe it's not a big deal is a job falls back to building its ISOs once in a while.

Reply:
We can further discuss this issue in the next PR where this script will be used.

Reply:
I reimplemented the usage of `last` tag according to Doug's suggestion.
Let's review and repost comment if any.